ui/components/app/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js

```diff
@@ -14,8 +15,20 @@ const ConfirmPageContainerSummary = (props) => {
     identiconAddress,
     nonce,
     origin,
+    type,
   } = props;
 
+  let hideTitle = false;
```

Comment:
Small stylistic comments on this section of code. We try to avoid an `if` block when it is simple and straightforward to do so. If this case, `hideTitle` can just be a `const` set to the logicial statement inside the `if ()` brackets.

Also, we try to use the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) when it can simplify code.

So this section of code could be simplified to

`const hideTitle = (type === TRANSACTION_TYPES.CONTRACT_INTERACTION || type === TRANSACTION_TYPES.DEPLOY_CONTRACT) && titleComponent.props?.value === '0x0'

Reply:
I also think it might be best if this logic is higher up in the component tree, and we pass `hideTitle` down as a prop. Ideally this component would be free of logic related to state. This wasn't the case even before this PR, but it is a direction we want to move.

You could use `currentTransaction` in `ui/components/app/confirm-page-container/confirm-page-container.component.js` to get the value and put this logic in that component. Then pass `hideTitle` down

Reply:
@danjm  Missed this last comment so I am adding one more commit with appropriate changes.